lib/model/unittest/CTokenListDataCategorizerTest.cc

```diff
@@ -605,46 +623,46 @@ BOOST_FIXTURE_TEST_CASE(testPreTokenised, CTestFixture) {
     BOOST_REQUIRE_EQUAL(ml::model::CLocalCategoryId{2},
                         categorizer.computeCategory(false, "<ml13-4608.1.p2ps: Info: > Source ML_SERVICE2 on 13122:867 has started.",
                                                     500));
-    BOOST_REQUIRE_EQUAL(ml::model::CLocalCategoryId{3},
+    BOOST_REQUIRE_EQUAL(ml::model::CLocalCategoryId{2},
                         categorizer.computeCategory(false, "<ml00-4201.1.p2ps: Info: > Service CUBE_CHIX, id of 132, has started.",
                                                     500));
```

Comment:
This is a case that's _not_ as good following the changes.  Previously it was nice that the extra weight on "Service" and "Source" meant that we differentiated services starting from sources starting.  After the adjacency weighting changes we only carried on doing that in this unit test because the token "has" was considered not to match between the two messages (because it has the higher adjacency weighting in the second message but not in the first).  Now that "has" is considered a match, that plus the massively high weighted "started" puts the two messages in the same category.

It's not ideal but I think the changes of this PR are more justifiable than what we had from #1903.

In the long term I think we should try to do the following:

1. Have the tokenizer do the weighting rather than the categorizer - this will facilitate 2 and 3
2. Only give higher weighting to adjacent dictionary words if they were separated by whitespace in the original message, not discarded tokens
3. Having decided we have a sufficiently long run of adjacent dictionary words, give higher weighting to all of them, not just the 3rd onwards - this will be fairer when one of the first two words is important (like "Source" vs "Service" in this case